diff --git a/package/Config.in b/package/Config.in
index e4eb2b7434..fb7ce50a03 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -2682,4 +2682,8 @@ menu "Text editors and viewers"
 	source "package/vim/Config.in"
 endmenu
 
+menu misc
+	source "package/hello_world/Config.in"
+endmenu
+
 endmenu
diff --git a/package/hello_world/Config.in b/package/hello_world/Config.in
new file mode 100644
index 0000000000..9d42da8c7a
--- /dev/null
+++ b/package/hello_world/Config.in
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_HELLO_WORLD
+	bool "Hello World"
+	help
+        Hello world package.
+
+        http://example.com
\ No newline at end of file
diff --git a/package/hello_world/hello_world.mk b/package/hello_world/hello_world.mk
new file mode 100644
index 0000000000..cc63f3b565
--- /dev/null
+++ b/package/hello_world/hello_world.mk
@@ -0,0 +1,19 @@
+################################################################################
+#
+# hello_world
+#
+################################################################################
+
+HELLO_WORLD_VERSION = 1.0
+HELLO_WORLD_SITE = ./package/hello_world/src
+HELLO_WORLD_SITE_METHOD = local
+
+define HELLO_WORLD_BUILD_CMDS
+    $(MAKE) CC="$(TARGET_CC)" LD="$(TARGET_LD)" -C $(@D)
+endef
+
+define HELLO_WORLD_INSTALL_TARGET_CMDS
+	$(INSTALL) -D -m 0755 $(@D)/hello_world $(TARGET_DIR)/usr/bin
+endef
+
+$(eval $(generic-package))
diff --git a/package/hello_world/src/.gitignore b/package/hello_world/src/.gitignore
new file mode 100644
index 0000000000..2b7e585e09
--- /dev/null
+++ b/package/hello_world/src/.gitignore
@@ -0,0 +1 @@
+hello_world
\ No newline at end of file
diff --git a/package/hello_world/src/Makefile b/package/hello_world/src/Makefile
new file mode 100644
index 0000000000..22b557c515
--- /dev/null
+++ b/package/hello_world/src/Makefile
@@ -0,0 +1,9 @@
+CC = gcc
+
+.PHONY: clean
+
+hello_world: hello_world.c
+	$(CC) -o '$@' '$<'
+
+clean:
+	rm hello_world
diff --git a/package/hello_world/src/hello_world.c b/package/hello_world/src/hello_world.c
new file mode 100644
index 0000000000..bcd30fbf99
--- /dev/null
+++ b/package/hello_world/src/hello_world.c
@@ -0,0 +1,5 @@
+#include <stdio.h>
+
+int main(void) {
+    puts("Hello world from Buildroot!");
+}
